src/components/product_list.jsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useRef, useState } from "react";
+import { useSelector } from "react-redux";
+
+import useIntersectionObserver from "hooks/use_intersection_observer";
+
+import Item from "components/product_item";
+import { Tray } from "styles/tray";
+
+const ProductList = ({ product_list }) => {
+  const page = useRef(4);
+  const [viewList, setList] = useState([
+    ...product_list.slice(0, page.current * 4),
+  ]);
+  const target = useRef(null);
+
+  const [observe, unobserve] = useIntersectionObserver(() => {
+    setList([...product_list.slice(0, (page.current + 1) * 4)]);
+    page.current = page.current + 1;
+  });
```

Comment:
```suggestion
  const [observe, unobserve] = useIntersectionObserver(() => {
    setList([...product_list.slice(0, (page.current + 1) * 4)]);
    page.current = page.current + 1;
  });
```

함수로 만들어서 전달해줄 것 같아요.